comictaggerlib/main.py

```diff
@@ -15,226 +15,144 @@
-    else:
-        os.environ["QtWidgets.QT_AUTO_SCREEN_SCALE_FACTOR"] = "1"
-        args = []
-        if opts.darkmode:
-            args.extend(["-platform", "windows:darkmode=2"])
-        args.extend(sys.argv)
-        app = Application(args)
-
-        # needed to catch initial open file events (macOS)
-        app.openFileRequest.connect(lambda x: opts.files.append(x.toLocalFile()))
-
-        if platform.system() == "Darwin":
-            # Set the MacOS dock icon
-            app.setWindowIcon(QtGui.QIcon(str(graphics_path / "app.png")))
-
-        if platform.system() == "Windows":
-            # For pure python, tell windows that we're not python,
-            # so we can have our own taskbar icon
-            import ctypes
-
-            myappid = "comictagger"  # arbitrary string
-            ctypes.windll.shell32.SetCurrentProcessExplicitAppUserModelID(myappid)  # type: ignore[attr-defined]
-            # force close of console window
-            swp_hidewindow = 0x0080
-            console_wnd = ctypes.windll.kernel32.GetConsoleWindow()  # type: ignore[attr-defined]
-            if console_wnd != 0:
-                ctypes.windll.user32.SetWindowPos(console_wnd, None, 0, 0, 0, 0, swp_hidewindow)  # type: ignore[attr-defined]
-
-        if platform.system() != "Linux":
-            img = QtGui.QPixmap(str(graphics_path / "tags.png"))
-
-            splash = QtWidgets.QSplashScreen(img)
-            splash.show()
-            splash.raise_()
-            QtWidgets.QApplication.processEvents()
-
-        try:
-            tagger_window = TaggerWindow(opts.files, settings, opts=opts)
-            tagger_window.setWindowIcon(QtGui.QIcon(str(graphics_path / "app.png")))
-            tagger_window.show()
-
-            # Catch open file events (macOS)
-            app.openFileRequest.connect(tagger_window.open_file_event)
-
-            if platform.system() != "Linux":
-                splash.finish(tagger_window)
-
-            sys.exit(app.exec())
-        except Exception:
-            logger.exception("GUI mode failed")
-            QtWidgets.QMessageBox.critical(
-                QtWidgets.QMainWindow(), "Error", "Unhandled exception in app:\n" + traceback.format_exc()
+            logger.exception("Failed to load publishers from %s", json_file)
+            # show_exception_box(str(e))
+
+
+class App:
+    """docstring for App"""
+
+    def __init__(self) -> None:
+        self.options: dict[str, dict[str, Any]] = {}
+        self.initial_arg_parser = settings.initial_cmd_line_parser()
+
+    def run(self) -> None:
+        opts = self.initialize()
+        self.register_options()
+        self.parse_options(opts.config)
+        self.initialize_dirs()
+
+        self.ctmain()
+
+    def initialize(self) -> argparse.Namespace:
+        opts, _ = self.initial_arg_parser.parse_known_args()
+        assert opts is not None
+        setup_logging(opts.verbose, opts.config.user_log_dir / "log")
```

Comment:
The `/ "log"` is causing the /log/log issue.

Reply:
Done